plugin/pkg/scheduler/algorithm/priorities/util/topologies.go

```diff
@@ -52,8 +52,12 @@ func PodMatchesTermsNamespaceAndSelector(pod *v1.Pod, affinityPod *v1.Pod, term
 	return true, nil
 }
 
-// nodesHaveSameTopologyKeyInternal checks if nodeA and nodeB have same label value with given topologyKey as label key.
-func nodesHaveSameTopologyKeyInternal(nodeA, nodeB *v1.Node, topologyKey string) bool {
+// NodesHaveSameTopologyKey checks if nodeA and nodeB have same label value with given topologyKey as label key.
+// Returns false if topologyKey is empty.
+func NodesHaveSameTopologyKey(nodeA, nodeB *v1.Node, topologyKey string) bool {
+	if len(topologyKey) == 0 {
```

Comment:
There's actually [this method](https://github.com/kubernetes/kubernetes/blob/55bee3ad21f025b1416a4e1f10de753f484b66d3/plugin/pkg/scheduler/algorithm/priorities/util/topologies.go#L63) of Topologies struct which is also named `NodesHaveSameTopologyKey` and which takes `failureDomains` into account (`failureDomains` there being the contents of `Topologies` struct `DefaultKeys` field, [see here](https://github.com/kubernetes/kubernetes/blob/082ce00ecaa2532f4395bc916c6f0eb482253cdc/plugin/pkg/scheduler/algorithm/priorities/interpod_affinity.go#L52)). The new function you mention is only used by pod affinity predicate which means we're dealing with `RequiredDuringScheduling`, not `PreferredDuringScheduling` and `--failure-domains` are not used, that's the current behaviour and I didn't change it. So this may not be immediately clear but my modifications don't change the behavior in any way. As of the new exported function, maybe I should change its name somehow to make it clear it should only be used when dealing with  `RequiredDuringScheduling` case?